### PR TITLE
feat(web): connect provider popover in the workflow editor

### PR DESCRIPTION
**Pull request by LetItRock**
### What change does this PR introduce?

When the Workflow Editor step doesn't have an active integration we want to show the popover which will allow the user to connect to the provider.

### Why was this change needed?

This PR is a part of improvements over the onboarding process.

### Other information (Screenshots)

Screenshot 2023-03-15 at 17 21 44

https://user-images.githubusercontent.com/2607232/225375541-9354002b-50d6-49da-b35e-6bdc642c0a5f.mov

### CODE REVIEW

File: apps/web/src/design-system/popover/Popover.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  content?: ReactNode;
+  title?: string;
+  description?: string;
+  url?: string;
+  urlText?: string;
+  onUrlClick?: MouseEventHandler<HTMLAnchorElement>;
+  titleGradient: 'red' | 'blue' | 'none';
+} & Omit<MantinePopoverProps, 'children'>;
+
+export const Popover = ({
```
Comment (LetItRock) [inline apps/web/src/design-system/popover/Popover.tsx:18]: moved this code from the `NodeStep` popovers shown on the Demo Workflow Tree to the design system as we will use similar popovers in other places

File: apps/web/src/design-system/template-button/ChannelButton.tsx [hunk 2]
```diff
@@ -26,7 +27,7 @@ interface ITemplateButtonProps {
   active?: boolean;
   action?: boolean;
   testId?: string;
-  tabKey: string;
+  tabKey?: ChannelTypeEnum;
```
Comment (LetItRock) [inline apps/web/src/design-system/template-button/ChannelButton.tsx:30]: fixed type

File: apps/web/src/design-system/template-button/ChannelButton.tsx [hunk 3]
```diff
@@ -39,6 +40,7 @@ interface ITemplateButtonProps {
   dragging?: boolean;
   setActivePage?: (string) => void;
   disabled?: boolean;
+  hasActiveIntegration: boolean;
```
Comment (LetItRock) [inline apps/web/src/design-system/template-button/ChannelButton.tsx:43]: the flag that says if the step has an active integration

File: apps/web/src/design-system/template-button/ChannelButton.tsx [hunk 4]
```diff
@@ -238,16 +248,30 @@ export function ChannelButton({
...
+                  Delete {isChannel ? 'Step' : 'Action'}
                 </Menu.Item>
               </Menu.Dropdown>
             </Menu>
           </When>
         </ActionWrapper>
       </ButtonWrapper>
-
-      {stepErrorContent && (
+      {!hasActiveIntegration && (
```
Comment (LetItRock) [inline apps/web/src/design-system/template-button/ChannelButton.tsx:258]: no active integration show the popover to connect with a provider

File: apps/web/src/design-system/template-button/ChannelButton.tsx [hunk 5]
```diff
@@ -238,16 +248,30 @@ export function ChannelButton({
...
+          target={<ErrorCircle data-test-id="error-circle" dark={theme.colorScheme === 'dark'} />}
+          title="Connect provider"
+          titleGradient="red"
+          description="Please configure a chat provider to send notifications over this channel"
+          content={
+            <ConfigureProviderButton onClick={toggleIntegrationsModalVisible}>Configure</ConfigureProviderButton>
+          }
+        />
+      )}
+      {hasActiveIntegration && stepErrorContent && (
```
Comment (LetItRock) [inline apps/web/src/design-system/template-button/ChannelButton.tsx:273]: if there is active integration and has error, then show the popover with error

File: apps/web/src/design-system/typography/label/index.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import styled from '@emotion/styled';
+
+export const Label = styled.div<{ gradientColor?: 'red' | 'blue' | 'none' }>`
```
Comment (LetItRock) [inline apps/web/src/design-system/typography/label/index.ts:3]: this one was used in other places around the code

File: apps/web/src/pages/templates/workflow/workflow/FlowEditor.tsx [hunk 7]
```diff
@@ -206,13 +207,20 @@ export function FlowEditor({
     step: IStepEntity,
     i: number
   ): Node {
+    const channel = getChannel(step.template.type);
+    let hasActiveIntegration = true;
+    if (channel?.tabKey !== ChannelTypeEnum.IN_APP) {
+      hasActiveIntegration = !!integrations?.some((integration) => integration.channel === channel?.tabKey);
+    }
```
Comment (LetItRock) [inline apps/web/src/pages/templates/workflow/workflow/FlowEditor.tsx:214]: for in-app we don't have integrations...

**Review by djabarovgeorge — APPROVED** [review]
Looks great to me :)

**Pull request merged**